x/xcvm/keeper/msg_server.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/notional-labs/composable/v6/x/xcvm/types"
+)
+
+var _ types.MsgServer = msgServer{}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{
+		Keeper: keeper,
+	}
+}
+
+type msgServer struct {
+	Keeper
+}
+
+func (k Keeper) SendTransferIntent(goCtx context.Context, msg *types.MsgSendTransferIntent) (*types.MsgSendTransferIntentResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	err := k.SendEthTransferIntent(ctx, msg)
+	if err != nil {
```

Comment:
nit: better to reduce scope of variables (same for others)
```suggestion
	if err := k.SendEthTransferIntent(sdk.UnwrapSDKContext(goCtx), msg); err != nil {
```